src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java

```diff
@@ -320,44 +875,656 @@ protected boolean isInsideWorkingBound(final AtlasEntity entity)
+     */
+    private boolean isSignificantGeometry(final Geometry geometry)
+    {
+        return geometry.isValid() && (geometry.getDimension() == 1
+                && geometry.getLength() > CountryBoundaryMap.LINE_BUFFER
+                || geometry.getDimension() == 2
+                        && geometry.getArea() > CountryBoundaryMap.AREA_BUFFER);
+    }
+
+    /**
+     * Given a slice for a Line entity, construct an appropriate PolyLine for the new sliced Line
+     * member. In the case of a Polygon (i.e. Area), the PolyLine will be constructed such that the
+     * winding represents the winding of the original entity
+     *
+     * @param slice
+     *            The slice for a Line entity
+     * @param line
+     *            The Line being sliced
+     * @return The best PolyLine to represent that slice for the Line
      */
-    protected boolean isOutsideWorkingBound(final Geometry geometry)
+    private PolyLine processSlice(final Geometry slice, final Line line)
     {
-        final String countryCode = CountryBoundaryMap.getGeometryProperty(geometry,
-                ISOCountryTag.KEY);
+        PolyLine polylineForSlice;
+        if (slice instanceof LineString)
+        {
+            polylineForSlice = JTS_POLYLINE_CONVERTER.backwardConvert((LineString) slice);
+        }
+        else if (slice instanceof org.locationtech.jts.geom.Polygon)
+        {
+            polylineForSlice = JTS_POLYLINE_CONVERTER
+                    .backwardConvert(((org.locationtech.jts.geom.Polygon) slice).getExteriorRing());
+        }
+        else
+        {
+            throw new CoreException("Unexpected geometry when slicing line {}",
+                    line.getIdentifier());
+        }
 
-        if (countryCode != null)
+        // JTS frequently reverses the winding during slicing-- this checks the winding of the
+        // original geometry, and reverse the winding of the slice if needed
+        if (line.isClosed() && slice instanceof org.locationtech.jts.geom.Polygon)
         {
-            return this.getCountries() != null && !this.getCountries().isEmpty()
-                    && !this.getCountries().contains(countryCode);
+            final boolean originalClockwise = new Polygon(line.asPolyLine().truncate(0, 1))
+                    .isClockwise();
+            final boolean sliceClockwise = new Polygon(polylineForSlice.truncate(0, 1))
+                    .isClockwise();
+            if (originalClockwise != sliceClockwise)
+            {
+                polylineForSlice = polylineForSlice.reversed();
+            }
         }
+        return polylineForSlice;
+    }
 
-        // Assume it's inside the bound
-        return false;
+    /**
+     * Given a multipolygon Relation, purge any members that don't meet the criteria in the OSM
+     * specification
+     *
+     * @param relation
+     *            The Relation to check
+     */
+    private void purgeInvalidMultiPolygonMembers(final CompleteRelation relation)
```

Comment:
More of a general question. What is the thinking behind removing bad member roles? Wouldn't that blind any future AtlasCheck for that specific error?
My initial gut feel is that even if some members are not inner nor outer, they could be ignored when slicing the relation. However if that makes the code significantly more complicated, I am all for keeping this as is.

Reply:
honestly, the justification for this is twofold. first off, we're going to ignore geometry from any of these anyway. but second, and more importantly, if we do this, we're guaranteed the important condition that this relation is *lowest order* which for a raw atlas very nearly means *line members only*. without that condition, we have handle the case that we've already sliced a child relation member, which gets into nested misery of changes for splitting this and finding which child split relation maps to this relation's new split relation. it's awful and not even conformant to the specification, so we just don't support it. 

there's nothing really stopping us from including the points or lines without valid roles, but if we're going so far as to filter these down to lowest order because of this, I feel like we should just go all out and make the relation fully conform to expectations.